gtsam/geometry/CameraSet.h

```diff
@@ -193,6 +193,105 @@ class CameraSet : public std::vector<CAMERA, Eigen::aligned_allocator<CAMERA> >
+
+    // here we have to deal with the fact that some blocks may share the same keys
+    if (nrUniqueKeys == nrNonuniqueKeys) {  // if there is 1 calibration key per camera
+      augmentedHessianUniqueKeys = SymmetricBlockMatrix(
+          dims, Matrix(augmentedHessian.selfadjointView()));
+    } else {  // if multiple cameras share a calibration we have to rearrange
+      // the results of the Schur complement matrix
+      std::vector<DenseIndex> nonuniqueDims(nrNonuniqueKeys + 1);  // this also includes the b term
+      std::fill(nonuniqueDims.begin(), nonuniqueDims.end() - 1, NDD);
+      nonuniqueDims.back() = 1;
+      augmentedHessian = SymmetricBlockMatrix(
+          nonuniqueDims, Matrix(augmentedHessian.selfadjointView()));
+
+      // get map from key to location in the new augmented Hessian matrix (the one including only unique keys)
+      std::map<Key, size_t> keyToSlotMap;
+      for (size_t k = 0; k < nrUniqueKeys; k++) {
+        keyToSlotMap[hessianKeys[k]] = k;
+      }
+
+      // initialize matrix to zero
+      augmentedHessianUniqueKeys = SymmetricBlockMatrix(
+          dims, Matrix::Zero(NDD * nrUniqueKeys + 1, NDD * nrUniqueKeys + 1));
+
+      // add contributions for each key: note this loops over the hessian with nonUnique keys (augmentedHessian)
+      // and populates an Hessian that only includes the unique keys (that is what we want to return)
+      for (size_t i = 0; i < nrNonuniqueKeys; i++) {  // rows
+        Key key_i = jacobianKeys.at(i);
+
+        // update information vector
+        augmentedHessianUniqueKeys.updateOffDiagonalBlock(
+            keyToSlotMap[key_i], nrUniqueKeys,
+            augmentedHessian.aboveDiagonalBlock(i, nrNonuniqueKeys));
+
+        // update blocks
+        for (size_t j = i; j < nrNonuniqueKeys; j++) {  // cols
+          Key key_j = jacobianKeys.at(j);
+          if (i == j) {
+            augmentedHessianUniqueKeys.updateDiagonalBlock(
+                keyToSlotMap[key_i], augmentedHessian.diagonalBlock(i));
+          } else {  // (i < j)
+            if (keyToSlotMap[key_i] != keyToSlotMap[key_j]) {
+              augmentedHessianUniqueKeys.updateOffDiagonalBlock(
+                  keyToSlotMap[key_i], keyToSlotMap[key_j],
+                  augmentedHessian.aboveDiagonalBlock(i, j));
+            } else {
+              augmentedHessianUniqueKeys.updateDiagonalBlock(
+                  keyToSlotMap[key_i],
+                  augmentedHessian.aboveDiagonalBlock(i, j)
+                      + augmentedHessian.aboveDiagonalBlock(i, j).transpose());
+            }
+          }
+        }
+      }
+      // update bottom right element of the matrix
+      augmentedHessianUniqueKeys.updateDiagonalBlock(
+          nrUniqueKeys, augmentedHessian.diagonalBlock(nrNonuniqueKeys));
+    }
+    return augmentedHessianUniqueKeys;
+  }
+
+  /**
+   * non-templated version of function above
+   */
+  static SymmetricBlockMatrix SchurComplementAndRearrangeBlocks_3_12_6(
```

Comment:
Why? `SchurComplementAndRearrangeBlocks<3,12,6>` reads better to me than `SchurComplementAndRearrangeBlocks_3_12_6`, and if you used fixed-size (or mixed fixed-dynamic) matrices the dimensions might be derivable as well...

Reply:
do you have an example to clarify how to derive the dimension?

Regarding "SchurComplementAndRearrangeBlocks<3,12,6>": it does not compile on my side and I get "error: 
reference to overloaded function could not be resolved; did you mean to call it?". Unfortunately, I was not able to get around that error.